pkg/version/generate/release_generate_test.go

```diff
@@ -0,0 +1,77 @@
+// +build release
+
+package main
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/weaveworks/pctl/pkg/version"
+)
+
+var _ = Describe("release tests", func() {
```

Comment:
this is testing a binary right? do you think we can build it and test same way we do other bins?

Reply:
so this is testing the `release_generate.go` file which will be used by the tag release automation similar to [here](https://github.com/weaveworks/eksctl/blob/main/.github/workflows/tag-release.yaml) to generate the release version. I have added the tests with the binary and version flag in the integration test file.

Reply:
That makes sense 👍 . This file still needs to change I think, because the integration test are testing `pctl --version`, which is calling an internal lib function. The goreleaser will call `go run pkg/version/generate/release_generate.go` which is a different binary from `pctl`, so this `pkg/version/generate/release_generate_test.go` still needs to be a proper integration test for this binary.

Reply:
I've added proper integration tests to check the binary 👍🏻

Reply:
noice! I think we are still missing the test which checks the file was successfully written, but i am fine with this for now.